homeassistant/components/tankerkoenig/coordinator.py

```diff
@@ -81,6 +84,25 @@ async def async_setup(self) -> None:
 
             self.stations[station_id] = station
 
+        entity_reg = er.async_get(self.hass)
+        for entity in er.async_entries_for_config_entry(
+            entity_reg, self.config_entry.entry_id
+        ):
+            if entity.unique_id.split("_")[0] not in self._selected_stations:
+                _LOGGER.debug("remove obsolet entity entry %s", entity.entity_id)
+                entity_reg.async_remove(entity.entity_id)
+
+        device_reg = dr.async_get(self.hass)
+        for device in dr.async_entries_for_config_entry(
+            device_reg, self.config_entry.entry_id
+        ):
+            if not any(
+                (ATTR_ID, station_id) in device.identifiers
+                for station_id in self._selected_stations
+            ):
+                _LOGGER.debug("remove obsolet device entry %s", device.name)
```

Comment:
```suggestion
                _LOGGER.debug("Removing obsolete device entry %s", device.name)
```

Also, not sure if using f strings is preferred here

Reply:
f-strings are not recommended for logger messages, to avoid maybe useless parsing efforts for a never shown logging message (_got this hint already multiple times_ 🙈 )